pulser/tests/test_sequence_sampler.py

```diff
@@ -31,14 +31,16 @@ def assert_nested_dict_equality(got, want: dict) -> None:
         for qty in want["Global"][basis]:
             np.testing.assert_array_equal(
                 got["Global"][basis][qty],
-                want["Global"][basis][qty],
+                # skip final 0 sample from comparison
+                want["Global"][basis][qty][:-1],
```

Comment:
This helper function is used elsewhere (in the SLM tests for example), and by the name it is expected to check the dicts without any logic assumption on the form.

I would rather keep it untouched, and copy patch it in the `assert_same_samples_as_sim()` function.

Or, add a bool kwarg `skip_last` defaulting to `False` and do

```suggestion
                (
                        want["Global"][basis][qty][:-1]
                        if skip_last
                        else want["Global"][basis][qty]
                ),
```

and call `assert_nested_dict_equality(got, want, skip_last=True)` in `assert_same_samples_as_sim()`.

Reply:
I think it is much better now! Thanks 😃